src/app/public/modules/autocomplete/autocomplete.component.ts

```diff
@@ -301,7 +301,12 @@ export class SkyAutocompleteComponent
     const isLongEnough = (searchText.length >= this.searchTextMinimumCharacters);
     const isDifferent = (searchText !== this.searchText);
 
-    this.searchText = searchText.trim();
+    this.searchText = searchText;
+
+    // Double check that search text is defined before trying to trim off whitespace
+    if (this.searchText) {
+      this.searchText.trim();
```

Comment:
Could we validate and sanitize the string value before assigning to `this.searchText`?

We're checking if the `searchText` is empty a few lines, above. Perhaps you could use that logic to confirm the string is not undefined before trimming? https://github.com/blackbaud/skyux-lookup/blob/94306c340e82cf559f8bc3fe3d7535175110b2bf/src/app/public/modules/autocomplete/autocomplete.component.ts#L295

Reply:
Oh. I got trigger happy. That `isEmpty` will short circuit so we don't even need this if statement on autocomplete. I'll remove it.

Reply:
Ok. I reworked the `isEmpty` logic slightly and removed the later call I had added.